nixos/modules/services/databases/postgresql.nix

```diff
@@ -279,6 +290,8 @@ in
      "/share/postgresql"
     ];
 
+    system.extraDependencies = lib.optional (cfg.checkConfig && pkgs.stdenv.hostPlatform == pkgs.stdenv.buildPlatform) configFileCheck;
```

Comment:
It's not good UX to silently skip the syntax check when cross-building a system and having `checkConfig` set to `true`. In that case an assertion should be thrown IMHO.

Reply:
I copied the semantics of `nix.checkConfig`. 

https://github.com/NixOS/nixpkgs/blob/release-20.03/nixos/modules/services/misc/nix-daemon.nix#L60

I don't find it good ux that this module throws an assertion error *by default* when cross compiling.